lib/jsonld-signatures.js

```diff
@@ -141,23 +141,23 @@ api.sign = util.callbackify(async function(input, options) {
   options = _addEmbeddedContextDocumentLoader(options);
 
   // instantiate a proofPurpose handler
-  const {purpose} = options;
+  const {purpose, purposeParameters} = options;
+  let proofPurposeHandler;
   if(purpose) {
-    const {proofPurpose} = purpose;
-    if(!proofPurpose) {
-      throw new Error('`purpose.proofPurpose` is required.');
+    const ProofPurposeHandler = api.proofPurposes.use(purpose);
+    if(!ProofPurposeHandler) {
+      throw new Error(`Unsupported proof purpose handler "${purpose}".`);
     }
-    const ProofPurpose = api.proofPurposes.use(proofPurpose);
-    if(!ProofPurpose) {
-      throw new Error(`Unsupported proof purpose handler "${proofPurpose}".`);
-    }
-    purpose.proofPurposeHandler = new ProofPurpose(injector);
+    proofPurposeHandler = new ProofPurposeHandler(injector);
   }
   // TODO: won't work with static analysis?
   // use signature suite
   //const Suite = require('./suites/' + algorithm);
   const Suite = suites[algorithm];
-  return new Suite(injector).sign(input, options);
+  return new Suite(injector).sign(
+    input,
+    Object.assign(
+      {}, options, {purpose, proofPurposeHandler, purposeParameters}));
```

Comment:
If this does the same thing as `Object.assign()` ... do it:

`{...options, purpose, proofPurposeHandler, purposeParameters}`

Reply:
Using the spread operator causes karma tests to fail, I can add the object spread operator polyfill to allow for what you suggested above.